docs/src/main/sphinx/functions/lambda.rst

```diff
@@ -15,7 +15,102 @@ Lambda expressions are written with ``->``::
     x -> CAST(x AS JSON)
     x -> x + TRY(1 / 0)
 
+Examples
+--------
+
+Obtain the squared elements of an array column::
+
+    SELECT numbers,
+           transform(numbers, n -> n * n) as squared_numbers
+    FROM (
+        VALUES
+            (ARRAY[1, 2]),
+            (ARRAY[3, 4]),
+            (ARRAY[5, 6, 7])
+    ) AS t(numbers);
+
+.. code-block:: text
+
+      numbers  | squared_numbers
+    -----------+-----------------
+     [1, 2]    | [1, 4]
+     [3, 4]    | [9, 16]
+     [5, 6, 7] | [25, 36, 49]
+    (3 rows)
+
+Besides the values of the array column being manipulated,
+other columns can be captured as well within the lambda expression.
+The following example provides a showcase of this feature
+for calculating the value of ``f(x) = ax + b`` linear function::
+
+    SELECT xvalues,
+           a,
+           b,
+           transform(xvalues, x -> a * x + b) as linear_function_values
+    FROM (
+        VALUES
+            (ARRAY[1,2], 10, 5),
+            (ARRAY[3,4], 4, 2)
+    ) AS t(xvalues, a, b);
+
+.. code-block:: text
+
+     xvalues | a  | b | linear_function_values
+    ---------+----+---+------------------------
+     [1, 2]  | 10 | 5 | [15, 25]
+     [3, 4]  |  4 | 2 | [14, 18]
+    (2 rows)
+
+
+Find the array elements containing at least one value greater than ``100``::
```

Comment:
All link to coalesce docs in this paragraph

Reply:
I think that `COALESCE` (same as `ARRAY`) is not really related to the context of this example query.
If you insist in having a link to `COALESCE`, I'd appreciate to have some guidance on how to fill it into the paragraph.